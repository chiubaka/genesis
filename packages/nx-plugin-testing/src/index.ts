export * from "./setup";
export * from "./testingWorkspace";
export * from "./verdaccio";
