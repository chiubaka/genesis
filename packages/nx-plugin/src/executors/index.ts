export * from "./nxPluginE2e";
