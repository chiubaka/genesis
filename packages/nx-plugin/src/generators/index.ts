export * from "./linting";
export * from "./preset";
