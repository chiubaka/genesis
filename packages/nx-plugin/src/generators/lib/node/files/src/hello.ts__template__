import crypto from "node:crypto";
import fs from "node:fs";
import os from "node:os";
import path from "node:path";

export const hello = "Hello, world!";

export function writeHelloWorld() {
  const randomSuffix = crypto.randomBytes(48).toString("base64url");
  const fileName = `hello-${randomSuffix}.txt`;

  const tmpDir = os.tmpdir();
  const filePath = path.join(tmpDir, fileName);

  // eslint-disable-next-line security/detect-non-literal-fs-filename
  fs.writeFileSync(filePath, hello);

  return filePath;
}

export class Greeter {
  public hello() {
    return hello;
  }
}
