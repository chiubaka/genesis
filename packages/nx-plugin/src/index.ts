export * from "./generators";
export * from "./utils";
