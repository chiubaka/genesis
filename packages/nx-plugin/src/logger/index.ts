import { Logger, LoggerOptions, LogLevel } from "./Logger";
import { WinstonLogger } from "./winston";

export const createLogger = (
  name: string,
  options: Partial<LoggerOptions> = {},
): Logger => {
  return new WinstonLogger({
    level: LogLevel.Info,
    ...options,
    output: {
      console: true,
      file: [
        {
          path: `${name}.log`,
        },
      ],
      ...options.output,
    },
  });
};

export const generatorLogger = createLogger("generator");
