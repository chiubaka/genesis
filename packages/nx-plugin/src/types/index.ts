export * from "./codecov";
export * from "./dockerCompose";
export * from "./npm";
export * from "./nx";
export * from "./verdaccio";
export * from "./yarn";
