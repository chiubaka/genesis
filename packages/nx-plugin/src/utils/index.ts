export * from "./addDependenciesToPackageJson";
export * from "./exec";
export * from "./getProjectDir";
export * from "./git";
export * from "./github";
export * from "./Project";
export * from "./projectNameCasings";
export * from "./tasks";
