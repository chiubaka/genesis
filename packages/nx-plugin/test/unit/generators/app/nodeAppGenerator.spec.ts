import { createTreeWithEmptyWorkspace } from "@chiubaka/nx-plugin-testing";
import { Tree } from "@nx/devkit";

import { nodeAppGenerator, Project } from "../../../../src";
import { fileMatchesSnapshot, nodeProjectTestCases } from "../../../cases";

describe("nodeAppGenerator", () => {
  let tree: Tree;
  let project: Project;
  let e2eProject: Project;

  const getProject = () => {
    return project;
  };

  beforeAll(() => {
    tree = createTreeWithEmptyWorkspace();
    project = new Project(tree, "node-app", "application");
    e2eProject = new Project(tree, "node-app-e2e", "e2e");
  });

  nodeProjectTestCases(getProject);

  beforeAll(async () => {
    await nodeAppGenerator(tree, {
      name: "node-app",
    });
  });

  it("generates a single app project", () => {
    const projectNames = tree.children(project.relativePath(".."));

    expect(projectNames).toEqual(["node-app"]);
  });

  // TODO: Node app generator doesn't yet implement E2E tests
  // eslint-disable-next-line jest/no-disabled-tests
  it.skip("generates a single E2E project", () => {
    const e2eProjectNames = tree.children(e2eProject.relativePath(".."));

    // For whatever reason, the Tree directory API doesn't seem 100% correct, so names
    // don't match here (we get `node-app-` instead of `node-app-e2e`) while they do
    // match in E2E tests.
    expect(e2eProjectNames).toHaveLength(1);
  });

  it("generates a main.ts file", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.srcPath("main.ts"))).toBe(true);
  });

  it("generates an assets dir", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.srcPath("assets"))).toBe(true);
  });

  it("generates a sample file that uses node APIs", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.srcPath("hello.ts"))).toBe(true);
    expect(tree).toHaveFileWithContent(project.srcPath("hello.ts"), "node:");
  });

  it("does not generate an index.ts file", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.srcPath("index.ts"))).toBe(false);
  });

  it("does not generate an app dir", () => {
    // eslint-disable-next-line security/detect-non-literal-fs-filename
    expect(tree.exists(project.srcPath("app"))).toBe(false);
  });

  describe("webpack.config.js", () => {
    it("generates a webpack.config.js file", () => {
      // eslint-disable-next-line security/detect-non-literal-fs-filename
      expect(tree.exists(project.path("webpack.config.js"))).toBe(true);
    });

    fileMatchesSnapshot("webpack.config.js", getProject, (project: Project) => {
      return project.path("webpack.config.js");
    });
  });
});
