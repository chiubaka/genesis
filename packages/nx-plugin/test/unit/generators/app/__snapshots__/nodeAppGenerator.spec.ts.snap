// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodeAppGenerator .eslintrc.json matches snapshot 1`] = `
{
  "extends": [
    "../../.eslintrc.json",
  ],
  "ignorePatterns": [
    "node_modules",
  ],
  "overrides": [
    {
      "files": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "parserOptions": {
        "project": [
          "packages/node-app/tsconfig.spec.json",
        ],
      },
    },
    {
      "excludedFiles": [
        "jest.config.ts",
        "test/**/*.ts",
        "*.spec.ts",
        "*.test.ts",
      ],
      "files": [
        "*.ts",
      ],
      "parserOptions": {
        "project": [
          "packages/node-app/tsconfig.app.json",
        ],
      },
    },
  ],
}
`;

exports[`nodeAppGenerator jest.config.ts matches snapshot 1`] = `
"export default {
  displayName: "node-app",
  preset: "../../jest.preset.js",
    globals: {
    "ts-jest": {
      tsconfig: "<rootDir>/tsconfig.spec.json",
    },
  },
      testEnvironment: "node",
    transform: {
        "^.+\\\\.[tj]s$": "ts-jest",
      },
    moduleFileExtensions: ["ts", "js", "html", "json"],
    coverageDirectory: "../../reports/coverage/packages/node-app",
  reporters: [
    "default",
    [
      "jest-junit",
      {
        addFileAttribute: "true",
        classNameTemplate: "{suitename}",
        outputDirectory: "reports/junit",
        outputName: "node-app.xml",
      },
    ],
  ],
};
"
`;

exports[`nodeAppGenerator project.json matches snapshot 1`] = `
{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "name": "node-app",
  "projectType": "application",
  "sourceRoot": "packages/node-app/src",
  "tags": [],
  "targets": {
    "build": {
      "configurations": {
        "development": {},
        "production": {},
      },
      "defaultConfiguration": "production",
      "executor": "@nx/webpack:webpack",
      "options": {
        "assets": [
          "packages/node-app/src/assets",
        ],
        "compiler": "tsc",
        "isolatedConfig": true,
        "main": "packages/node-app/src/main.ts",
        "outputPath": "dist/packages/node-app",
        "target": "node",
        "tsConfig": "packages/node-app/tsconfig.app.json",
        "webpackConfig": "packages/node-app/webpack.config.js",
      },
      "outputs": [
        "{options.outputPath}",
      ],
    },
    "lint": {
      "executor": "@nx/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "packages/node-app",
        ],
      },
      "outputs": [
        "{options.outputFile}",
      ],
    },
    "serve": {
      "configurations": {
        "development": {
          "buildTarget": "node-app:build:development",
        },
        "production": {
          "buildTarget": "node-app:build:production",
        },
      },
      "defaultConfiguration": "development",
      "executor": "@nx/js:node",
      "options": {
        "buildTarget": "node-app:build",
      },
    },
    "test": {
      "configurations": {
        "ci": {
          "ci": true,
          "codeCoverage": true,
        },
      },
      "executor": "@nx/jest:jest",
      "options": {
        "jestConfig": "packages/node-app/jest.config.ts",
        "passWithNoTests": true,
      },
      "outputs": [
        "reports/coverage/packages/node-app",
        "reports/junit/node-app.xml",
      ],
    },
  },
}
`;

exports[`nodeAppGenerator tsconfig.app.json or tsconfig.lib.json matches snapshot 1`] = `
{
  "compilerOptions": {
    "declaration": true,
    "types": [
      "node",
    ],
  },
  "exclude": [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "test",
  ],
  "extends": "./tsconfig.json",
  "include": [
    "**/*.ts",
  ],
}
`;

exports[`nodeAppGenerator tsconfig.json matches snapshot 1`] = `
{
  "compilerOptions": {
    "lib": [
      "es2022",
    ],
    "module": "commonjs",
    "outDir": "../../dist/out-tsc",
    "target": "es2022",
  },
  "extends": "../../tsconfig.base.json",
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.app.json",
    },
    {
      "path": "./tsconfig.spec.json",
    },
  ],
}
`;

exports[`nodeAppGenerator tsconfig.spec.json matches snapshot 1`] = `
{
  "compilerOptions": {
    "types": [
      "jest",
      "node",
    ],
  },
  "extends": "./tsconfig.json",
  "include": [
    "jest.config.ts",
    "**/*.spec.ts",
    "**/*.test.ts",
    "**/*.d.ts",
    "test",
  ],
}
`;

exports[`nodeAppGenerator webpack.config.js webpack.config.js matches snapshot 1`] = `
"const { composePlugins, withNx } = require('@nx/webpack');

// Nx plugins for webpack.
module.exports = composePlugins(withNx(), (config) => {
  // Update the webpack config as needed here.
  // e.g. \`config.plugins.push(new MyPlugin())\`
  return config;
});
"
`;
